### PR TITLE
En 4630 jayasimha

### PR DESCRIPTION
**Pull request by jayasimha-raghavan-unskript**
## Description
Please include a summary of the change, motivation and context.
[EN-4630] 

This PR brings in support for statefulset to our Helm chart.

### Testing
Please describe the tests that you ran to verify your changes. Please summarize what did you test and what needs to be tested e.g. deployed and tested helm chart locally. 

#### Helm Install Command output
```
(base) [Awesome-CloudOps-Automation]- (EN-4630-jayasimha)$helm upgrade --install --namespace awesome-ops --create-namespace awesome-ops ./helm 
Release "awesome-ops" does not exist. Installing it now.
W0607 10:42:42.045118   20296 warnings.go:70] Autopilot increased resource requests for StatefulSet awesome-ops/awesome-ops-awesome-runbooks to meet requirements. See http://g.co/gke/autopilot-resources
NAME: awesome-ops
LAST DEPLOYED: Wed Jun  7 10:42:41 2023
NAMESPACE: awesome-ops
STATUS: deployed
REVISION: 1
TEST SUITE: None
NOTES:
1. Get the application URL by running these commands:
  export POD_NAME=$(kubectl get pods --namespace awesome-ops -l "app.kubernetes.io/name=awesome-runbooks,app.kubernetes.io/instance=awesome-ops" -o jsonpath="{.items[0].metadata.name}")
  export CONTAINER_PORT=$(kubectl get pod --namespace awesome-ops $POD_NAME -o jsonpath="{.spec.containers[0].ports[0].containerPort}")
  echo "Visit http://127.0.0.1:8888 to use your application"
  kubectl --namespace awesome-ops port-forward $POD_NAME 8888:$CONTAINER_PORT

```
#### PODS status
```
(base) [Awesome-CloudOps-Automation]- (EN-4630-jayasimha)$kubectl get pods -n awesome-ops                                   
NAME                             READY   STATUS    RESTARTS   AGE
awesome-ops-awesome-runbooks-0   0/1     Running   0          31s
(base) [Awesome-CloudOps-Automation]- (EN-4630-jayasimha)$
```

#### Running state
Screenshot 2023-06-07 at 10 59 39 AM

### Checklist:
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] Any dependent changes have been merged and published. 

### Documentation
Make sure that you have documented corresponding changes in this repository. 

[EN-4630]: https://unskript.atlassian.net/browse/EN-4630?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: helm/templates/deployment.yaml [hunk 1]
```diff
@@ -1,30 +1,31 @@
...
 kind: Deployment
 metadata:
-  name: {{ include "unskript-awesome-runbooks.fullname" . }}
+  name: {{ include "awesome-runbooks.fullname" . }}
   labels:
-    {{- include "unskript-awesome-runbooks.labels" . | nindent 4 }}
+    {{- include "awesome-runbooks.labels" . | nindent 4 }}
 spec:
-  {{- if not .Values.autoscaling.enabled }}
+  {{- if not (eq .Values.autoscaling.enabled "") }}
```
Comment (amit-chandak-unskript) [inline helm/templates/deployment.yaml:9]: not needed

File: helm/values.yaml [hunk 2]
```diff
@@ -28,6 +28,39 @@ podAnnotations: {}
 podSecurityContext: {}
   # fsGroup: 2000
 
+# Flag to indicate whether to use stateful-set or not
+useStatefulSet: true 
+
+common:
+  namespace: "awesome-ops"
```
Comment (amit-chandak-unskript) [inline helm/values.yaml:35]: Put a comment for every field

File: helm/values.yaml [hunk 3]
```diff
@@ -28,6 +28,39 @@ podAnnotations: {}
...
+  annotations: {}
+  mountPath: /unskript
+  subPath: ""
+  volumeClaimTemplate:
+    selector: {} 
+    requests: {}
+    dataSource: {}
+
+
+replicaSet:
```
Comment (amit-chandak-unskript) [inline helm/values.yaml:61]: dont need this

File: helm/templates/deployment.yaml (hunk 1 above)
Comment (jayasimha-raghavan-unskript) [reply helm/templates/deployment.yaml:9]: Done

File: helm/values.yaml [hunk 4]
```diff
@@ -40,22 +73,23 @@ service:
   type: ClusterIP
   port: 8888
 
-resources: {}
+resources:
   # We usually recommend not to specify default resources and to leave this as a conscious
   # choice for the user. This also increases chances charts run on environments with little
   # resources, such as Minikube. If you do want to specify resources, uncomment the following
   # lines, adjust them as necessary, and remove the curly braces after 'resources:'.
```
[the remark this answers is not in the document]
Comment (jayasimha-raghavan-unskript) [reply helm/values.yaml]: Done

File: helm/values.yaml (hunk 2 above)
Comment (jayasimha-raghavan-unskript) [reply helm/values.yaml:38]: Done

**Review by amit-chandak-unskript — APPROVED** [review]

**Pull request merged**